main/test/CMakeLists.txt

```diff
@@ -3,12 +3,13 @@ set(UNIT_TESTS
         init/grid.cpp
         init/isobaric_cube.cpp
         io/arg_parser.cpp
+        observables/gravitational_waves.cpp
         test_main.cpp)
 
 set(exename frontend_units)
 add_executable(${exename} ${UNIT_TESTS})
 target_compile_options(${exename} PRIVATE -Wall -Wextra -Wno-unknown-pragmas)
 
-target_include_directories(${exename} PRIVATE ${CSTONE_DIR} ${PROJECT_SOURCE_DIR}/main/src)
-target_link_libraries(${exename} PRIVATE GTest::gtest_main)
+target_include_directories(${exename} PRIVATE ${CSTONE_DIR} ${SPH_DIR} ${PROJECT_SOURCE_DIR}/main/src)
+target_link_libraries(${exename} PRIVATE GTest::gtest_main ${MPI_CXX_LIBRARIES})
```

Comment:
We don't enable MPI in unit tests because multi-rank programs are harder to debug, and, on HPC systems can't be run on login nodes. If you really need MPI, you need to create a standalone test executable and
in that case, you're not really writing a unit test anymore. You're combining multiple units and at that point what you're writing is an integration test.

What you need is a unit test for `d2QuadpoleMomentum` and another one for the `httplus-cross` function that you're going to create. This doesn't need the dependency on ParticlesData and doesn't need MPI and would give you full code coverage except for the reduction call. It's probably not needed, but if you then still want, you can write an MPI-aware integration test that actually tests the MPI reduction. This is something that your test does not do, if you execute it on more than one rank, it will fail.